iGCS/UserInterface/ViewControllers/WaypointSettingsViewController.m

```diff
@@ -0,0 +1,175 @@
+#import "SettingsWaypointCell.h"
+
+#import "SettingsData.h"
+
+
+
+@interface WaypointSettingsViewController ()
+
+@property (nonatomic, retain) NSArray *generalSettingsArray;
+@property (nonatomic, retain) NSArray *otherSettingsArray;
+
+@property (nonatomic, retain) NSMutableArray *sectionKeysArray;
+@property (nonatomic, retain) NSMutableDictionary *sectionContentsDict;
+
+//@property (nonatomic, assign) NSInteger altitude;
+
+
+@end
+
+@implementation WaypointSettingsViewController
+
+- (instancetype)initWithStyle:(UITableViewStyle)style {
+    self = [super initWithStyle:style];
+    if (self) {
+        // Custom initialization
+    }
+    return self;
+}
+
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self createSectionData];
+    
+    // Uncomment the following line to preserve selection between presentations.
+    // self.clearsSelectionOnViewWillAppear = NO;
+    
+    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
+    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
+}
+
+
+- (void) createSectionData {
+    
+    //general settings
+    self.generalSettingsArray = @[@"altitude", @"radius", @"ceiling"];
+    
+    
+    NSMutableArray *keys = [[NSMutableArray alloc] init];
+    NSMutableDictionary *contents = [[NSMutableDictionary alloc] init];
+    
+    NSString *generalSectionKey = @"WaypointSettings";
+   
+    
+    [contents setObject:self.generalSettingsArray forKey:generalSectionKey];
+    
+    [keys addObject:generalSectionKey];
+    
+    [self setSectionKeysArray:keys];
+    [self setSectionContentsDict:contents];
+}
+
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return [[self sectionKeysArray] count];
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    NSString *key = [self.sectionKeysArray objectAtIndex:section];
+    NSArray *contents = [self.sectionContentsDict objectForKey:key];
+    NSInteger rows = contents.count;
+    return rows;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    
+    static NSString *cellIdentifier = @"Cell";
+    SettingsWaypointCell *cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier];
+    if (cell == nil) {
+        cell = [[SettingsWaypointCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellIdentifier];
+    }
+    NSString *key = [self.sectionKeysArray objectAtIndex:indexPath.section];
+    NSArray *contents = [self.sectionContentsDict objectForKey:key];
+    NSString *cellContent = [contents objectAtIndex:indexPath.row];
+    cell.textLabel.text = cellContent;
+    
+    if ([SettingsData sharedSettingsData].unitType == metric) {
```

Comment:
You could push this logic to the GCSSettingsModel with a NSString property and getter called `unitSystemName` or something that makes sense and returns the correct name based on the models current unitType setting.